resources/2024-10-dapr-bindings.md

```diff
@@ -0,0 +1,368 @@
+  - Production, AWS: Creates [CRD](https://docs.dapr.io/reference/components-reference/supported-bindings/s3/) only.
+
+- **Mailing Binding Recipe**
+  - Local Development: Creates both [CRD](https://docs.dapr.io/reference/components-reference/supported-bindings/smtp/) and a [MailDev](https://github.com/maildev/maildev) instance.
+  - Production: Creates [CRD](https://docs.dapr.io/reference/components-reference/supported-bindings/smtp/) only.
+
+### Detailed Design
+
+This design will require adding a new type to the Applications.Dapr resource provider. This includes:
+- Extending the TypeSpec schema definition to include the new Application.Dapr/Bindings resource
+- Adding the corresponding DaprBinding type to the internal representation (DaprRP)
+- Allowing a new type of Dapr component (binding) to be emitted in the backend Kubernetes cluster.
+
+#### Advantages (of each option considered)
+
+The main advantage of this implementation approach is that it is purely additive and will be non-breaking for existing users.
+
+#### Disadvantages (of each option considered)
+
+The main disadvantage of this approach is that it will lead to some code duplication.
+
+For each new Dapr Building Block type, we will need to create both a versioned/unversioned API type converter and a dedicated processor.
+
+The processor code for Dapr building blocks is largely similar across different types, with the only variation being the resource type involved.
+
+Example:
+- [Pub/Sub processor](https://github.com/radius-project/radius/blob/main/pkg/daprrp/processors/pubsubbrokers/processor.go)
+- [Configuration store processor](https://github.com/radius-project/radius/blob/main/pkg/daprrp/processors/configurationstores/processor.go)
+
+Similarly, the converter code is nearly identical for all Dapr building blocks, differing only in the resource type being converted.
+
+Example:
+- [Pub/Sub converter](https://github.com/radius-project/radius/blob/main/pkg/daprrp/api/v20231001preview/pubsubbroker_conversion.go)
+- [Configuration store converter](https://github.com/radius-project/radius/blob/main/pkg/daprrp/api/v20231001preview/configurationstore_conversion.go)
+
+#### Proposed Option
+
+This option is the most straightforward and efficient in terms of development time. Since the Dapr Building Block implementation is not yet complete, it would be better to have a simple implementation that can be improved later, even if it introduces some code duplication. Additionally, future implementations of User-Defined Types may lead to a refactor of the entire Applications.Dapr resource provider.
+
+### API design (if applicable)
+
+Aside from the new type, the API will remain identical to the other Dapr Building Blocks.
+
+**typespec/Applications.Dapr/bindings.tsp**
+```TypeSpec
+model DaprBindingResource
+  is TrackedResourceRequired<DaprBindingProperties, "DaprBindings"> {
```

Comment:
If the type name is `bindings` then this should be `TrackedResourceRequired<DaprBindingProperties, "bindings">`